### PR TITLE
Port WNDCLASS

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Ports `WNDCLASS` and style definitions. Convert `NMHDR`.

Port `LoadCursor`, `LoadIcon`, `ShowCursor`, and `DestroyWindow`.

Ports `CreateWindowEx` and manually implments `AsAny` marshalling. Note that arbitrary arrays are not supported in this change, but simple arrays of intrinsic types still are (`int[]`).

Adds `IHandle<HWND>` to `HWND` to reduce the need for multiple overloads (see `IsChild` and `SetParent`).

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7573)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/HWND.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Windows.Win32.Foundation
+{
+    internal partial struct HWND : IHandle<HWND>
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/HWND.cs:7]: @lonitra, @KlausLoeffelmann, @fernandanavamoya here is the self-implementation of `IHandle<T>`. Allows us to only write one overload.

**Review by lonitra — APPROVED** [review]
LGTM!

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ObjectSelectorEditor.Selector.cs [hunk 2]
```diff
@@ -211,8 +211,8 @@ protected unsafe override void WndProc(ref Message m)
 
                         break;
                     case User32.WM.REFLECT_NOTIFY:
-                        User32.NMHDR* nmtv = (User32.NMHDR*)(nint)m.LParamInternal;
-                        if (nmtv->code == (int)ComCtl32.NM.CLICK)
+                        NMHDR* nmtv = (NMHDR*)(nint)m.LParamInternal;
+                        if ((int)nmtv->code == (int)ComCtl32.NM.CLICK)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ObjectSelectorEditor.Selector.cs:215]: Is `nmtv->code` now a `uint`?

**Review by RussKie — APPROVED** [review]
👍

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ObjectSelectorEditor.Selector.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ObjectSelectorEditor.Selector.cs:215]: Yep.

**Pull request merged**